azurerm/resource_arm_sql_database.go

```diff
@@ -354,6 +468,19 @@ func resourceArmSqlDatabaseRead(d *schema.ResourceData, meta interface{}) error
 		return fmt.Errorf("Error making Read request on Sql Database %s: %+v", name, err)
 	}
 
+	oldThreatDetection, hasOldThreatDetection := d.GetOk("threat_detection_policy")
+	if hasOldThreatDetection {
```

Comment:
The problem was that the thread detection problem is a separate API call which always returns the default values, causing every `plan` operation to be the following every time, if the `threat_detection_policy` was not set in the manifest as it filled out the list in the state with all default values:

```
  ~ azurerm_sql_database.test
      threat_detection_policy.#: "1" => "0"
```

I've changed this so that the list element is marked `computed`, which alleviates the problem. Though I'm not sure if this is ok or if there is a better option to handle this case.